application/backend/src/main/java/cmpe451/group12/beabee/common/service/UserService.java

```diff
@@ -118,4 +120,66 @@ public UserAnalyticsDTO getAnalytics(Long user_id) {
     public UserGetDTO getUser(Long id) {
         return userMapper.mapToDto(userRepository.findById(id).orElseThrow(()-> new ResponseStatusException(HttpStatus.NOT_FOUND,"User not found!")));
     }
+
+    public MessageResponse followUser(Long userId, Long targetUserId) {
+        Users user = userRepository.findById(userId)
+                .orElseThrow(EntityNotFoundException::new);
+
+        Users targetUser = userRepository.findById(targetUserId)
+                .orElseThrow(EntityNotFoundException::new);
+
+        if (user.getFollowing().stream().anyMatch(following -> following.getUser_id().equals(targetUser.getUser_id()))) {
+            return new MessageResponse("Already following!", MessageType.ERROR);
+        }
+
+        user.getFollowing().add(targetUser);
+        targetUser.getFollowers().add(user);
+
+        userRepository.save(user);
+        userRepository.save(targetUser);
+
+        activityStreamService.followUserSchema(user, targetUser);
+
+        return new MessageResponse("User followed successfully!", MessageType.SUCCESS);
+    }
+
+    public MessageResponse unfollowUser(Long userId, Long targetUserId) {
+        Users user = userRepository.findById(userId)
+                .orElseThrow(EntityNotFoundException::new);
+
+        Users targetUser = userRepository.findById(targetUserId)
+                .orElseThrow(EntityNotFoundException::new);
+
+        if (user.getFollowing().stream().noneMatch(following -> following.getUser_id().equals(targetUser.getUser_id()))) {
+            return new MessageResponse("Not following!", MessageType.ERROR);
+        }
+
+        user.getFollowing().remove(targetUser);
+        targetUser.getFollowers().remove(user);
+
+        userRepository.save(user);
+        userRepository.save(targetUser);
+
+        activityStreamService.unfollowUserSchema(user, targetUser);
```

Comment:
There is no UNFOLLOW activity type in activity stream vocab: https://www.w3.org/TR/activitystreams-vocabulary/#activity-types . Either remove the unfollowUserSchema or change activityType to "REMOVE".

Reply:
Removing it for now